### PR TITLE
MLDB-1058: avoid allocating a std::string each time a ML::Parse_Conte…

### PR DESCRIPTION
**Pull request by wsourdeau**
…xt is instantiated

### CODE REVIEW

**Comment by jeremybarnes** [comment]
-1.  This will make the error messages much less helpful, as they won't tell us what the expression we were parsing was.  I agree with the end but not the means.  Can we do it a different way?

**Comment by wsourdeau** [comment]
@jeremybarnes for case such as in sql_expressoin.cc, I understand, but in other cases it may not be relevant. The use of "NO_FILE" is still optional. Would it be ok for you if I reverted sql_expression.cc?

**Comment by jeremybarnes** [comment]
No, the Parse_Context's error messages are a big part of the usability in all the places that they are used.  You're making a big change here when we only need a small one.  If a user of Parse_Context wants to have no filename, then it can do so externally, but that should be enabled at the caller.

**Comment by nicolaskruchten** [comment]
What's the status of this PR? What's the performance improvement like from this?

**Comment by wsourdeau** [comment]
@nicolaskruchten the technical aspect of the comments above have been discussed and settled with Jeremy before christmas. Benchmarks under way...

**Comment by nicolaskruchten** [comment]
We still need the +1 from @jeremybarnes to merge.

**Comment by wsourdeau** [comment]
@nicolaskruchten @jeremybarnes gain = 5kqps

File: base/parse_context.cc [hunk 1]
```diff
@@ -28,6 +28,8 @@ namespace ML {
 /* PARSE_CONTEXT                                                             */
 /*****************************************************************************/
 
+const std::string Parse_Context::NO_FILE("-");
```
Comment (jeremybarnes) [inline base/parse_context.cc]: Should be "CONSOLE", not NO_FILE.

**Comment by jeremybarnes** [comment]
With the renaming I just proposed, this is good to merge.

**Pull request merged**